pkg/apis/networking/v1/types.go

```diff
@@ -260,7 +260,8 @@ type RemoteSubnetSpec struct {
 	Range       AddressRange `json:"range"`
 	Type        NetworkType  `json:"type"`
 	ClusterName string       `json:"clusterName"`
-	TunnelNetID *uint32      `json:"tunnelNetID"`
+	// If the corresponding subnet is in underlay network, it will be set to nil
```

Comment:
It should not be nil for a remote underlay subnet either, because a pod from local overlay subnet need to access remote underlay subnet through a tunnel.

Reply:
> If it is nil, use local vxlan net id.

For this case, local vxlan net id is the tunnelNetID, maybe tunnelNetID should not be nil.

Reply:
Do not hide information from user if there is already a filed to show it. Just show it as directly as possible.

Reply:
> > If it is nil, use local vxlan net id.
> 
> For this case, local vxlan net id is the vxlan id used to tunneling, maybe tunnelNetID should not be nil either.

you mean the control plane should perceive the local vxlan net id? maybe too coupled?

Reply:
> > > If it is nil, use local vxlan net id.
> > 
> > 
> > For this case, local vxlan net id is the vxlan id used to tunneling, maybe tunnelNetID should not be nil either.
> 
> you mean the control plane should perceive the local vxlan net id? maybe too coupled?

You don't need to make it separated, whether use local vxlan net id or not, it's an intact logic. You always need to looking for the remote overlay network, the only thing you need to do is to look back to local overlay network if the remote overlay network not exist.

From another side, tunnelNetID is the actual vxlan id for tunneling, this is a good information for network operation engineer, we should not just make it nil if we have an vxlan id for it, whether the vlxan id is local vxlan id or not.